### ISSUE TITLE
Keyboard navigation in new history: dataset list

### ISSUE THREAD

**Issue by elichad** [galaxyproject/galaxy#13936]
**Describe the bug**

*Edit: oops, I thought I was testing on dev but I was actually on 22.01. Looks like a bunch of these have been fixed, yay*

- [Elements should not have tabindex greater than zero](https://dequeuniversity.com/rules/axe/4.4/tabindex) - all the dataset items violate this principle and it makes the page confusing to navigate. The datasets should not be the first thing a keyboard user encounters on the page, and the individual dataset buttons ('View data' etc) are so far down the tab order relative to the data set list (the rest of the page is in between!) that I thought I just could not get to them with a keyboard.
- I expected that pressing Space on a dataset item would expand the dataset, but instead it activated the selection mode and selected the dataset. These should be two separate controls. EDIT: actually, looks like the first time you reach the dataset in the tab order, Space does expand it. Second time around, it activates the checkbox but _the focus looks exactly the same_ (but I was jumping past the first instance by manually clicking buttons in the history panel). Point stands that this is just really confusing behaviour.
- Speaking of the selection mode, it's not clear how to deactivate it once it is activated. Pressing Space a second time deselected the checkbox, but the checkboxes themselves did not vanish. Pressing 'Select/deselect all' twice works, but tabbing to this is tedious at best and difficult at worst (especially with a funky tab order)

**Galaxy Version and/or server at which you observed the bug**
Galaxy Version: ~dev~ 22.01
~Commit: 2f01b32b948a85e0ce5ef916aa7b66597ab94371~

**To Reproduce**
1. Activate new history
2. Click a button in the history panel (Filter, etc)
3. Tab into dataset list
4. Press space
5. See checkbox activated
6. Press space again
7. See checkbox deactivated but still visible

**Expected behavior**
Pressing space expands/collapses the dataset

**Comment by dannon** [comment]
Keyboard navigation now works well in the history.  Thanks for the report!
